staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/conversion.go

```diff
@@ -352,13 +354,17 @@ func Convert_Slice_string_To_Slice_int32(in *[]string, out *[]int32, s conversio
 	return nil
 }
 
-// Convert_Slice_string_To_v1_DeletionPropagation allows converting a URL query parameter propagationPolicy
-func Convert_Slice_string_To_v1_DeletionPropagation(in *[]string, out *DeletionPropagation, s conversion.Scope) error {
+// Convert_Slice_string_To_Pointer_v1_DeletionPropagation allows converting a URL query parameter propagationPolicy
+func Convert_Slice_string_To_Pointer_v1_DeletionPropagation(in *[]string, out **DeletionPropagation, s conversion.Scope) error {
+	if in == nil {
+		return nil
+	}
+	str := ""
 	if len(*in) > 0 {
-		*out = DeletionPropagation((*in)[0])
-	} else {
-		*out = ""
+		str = (*in)[0]
```

Comment:
This looks fine to me, though I would like to get explicit ACK from Jordan that this is how we would like to handle [in!=nil && len(in) = 0]

Reply:
In practice, this method is only called with non-nil, non-zero-length slices, but we should remain consistent with previous behavior, so keep the `} else { ... = "" }` block.

On both master and this branch, query parameters have the following behavior:
* `?` -> nil
* `?propagationPolicy` -> &""
* `?propagationPolicy=` -> &""
* `?propagationPolicy=foo` -> &"foo"
* `?propagationPolicy=Orphan` -> &"Orphan"